File: SORTING/C++/insertionSort.cpp
#include <iostream>
using namespace std;

void insertionSort(int arr[], int n)
{
    int temp,j;
    for (int i = 1; i < n; i++)
    {
        temp = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > temp)
        {   // Moving the elements of array that are greater than temp, to one position ahead of their current position
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = temp;
    }
}

int main()
{
    int n;
    cout << "Enter the length of array:\n";
    cin >> n;
    int arr[n];
    cout << "Enter the elements of array:\n";
    for (int i = 0; i < n; i++)
        cin >> arr[i];     // Taking input
    insertionSort(arr, n); //Calling of insertionSort function
    cout << "Sorted array elements:\n";
    for (int i = 0; i < n; i++)
        cout << arr[i] << " "; //Printing out elements
    return 0;
}

File: algorithms/CSharp/src/Sorts/insertion-sort.cs
using System;

namespace Algorithms.Sorts
{
    public class InsertionSort
    {
        public static void Main()
        {
            int[] arr = { 800, 11, 50, 771, 649, 770, 240, 9 };
            Sort(arr);
            var result = string.Join(" ", arr);
            Console.WriteLine(result);
        }

        public static void Sort(int[] source)
        {
            int n = source.Length;
            for (int i = 1; i < n; ++i)
            {
                int key = source[i];
                int j = i - 1;

                // Move elements of arr[0..i-1], 
                // that are greater than key, 
                // to one position ahead of 
                // their current position 
                while (j >= 0 && source[j] > key)
                {
                    source[j + 1] = source[j];
                    j = j - 1;
                }
                source[j + 1] = key;
            }
        }
    }
}